openmetadata-ui/src/main/resources/ui/src/utils/Auth/TokenService/TokenServiceUtil.ts

```diff
@@ -105,6 +108,9 @@ class TokenService {
 
         return newToken;
       } else {
+        // Token doesn't need refreshing, clear the flag
+        this.clearRefreshInProgress();
+
```

Comment:
How it get reset for actual refresh?

Reply:
In The finally block (line 142) of fetchNewToken()